### PR TITLE
Ability to disable coordinated shutdown phases

### PR DESCRIPTION
**Pull request by johanandren**

### CODE REVIEW

File: akka-actor/src/main/resources/reference.conf [hunk 1]
```diff
@@ -1077,6 +1077,10 @@ akka {
     # - timeout=15s: Override the default-phase-timeout for this phase.
     # - recover=off: If the phase fails the shutdown is aborted
     #                and depending phases will not be executed.
+    # - enabled=off: Skip all tasks registered in this phase. DO NOT use
+    #                this to disable phases unless you are absolutely sure what the
+    #                consequences are. Many of the built in tasks depend on other tasks
+    #                having been executed in earlier phases and may break if those are disabled.
```
Comment (patriknw) [inline akka-actor/src/main/resources/reference.conf]: good

File: akka-actor/src/main/scala/akka/actor/CoordinatedShutdown.scala [hunk 2]
```diff
@@ -406,6 +406,12 @@ final class CoordinatedShutdown private[akka] (
       def loop(remainingPhases: List[String]): Future[Done] = {
         remainingPhases match {
           case Nil ⇒ Future.successful(Done)
+          case phase :: remaining if !phases(phase).enabled ⇒
+            tasks.get(phase) match {
+              case null  ⇒ // This pretty much is ok as there are no tasks
+              case tasks ⇒ log.warning("Phase [{}] disabled through configuration, skipping [{}] tasks", phase, tasks.size)
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/actor/CoordinatedShutdown.scala]: perhaps info is enough, they did it on purpose

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-actor/src/main/scala/akka/actor/CoordinatedShutdown.scala (hunk 2 above)
Comment (johanandren) [reply akka-actor/src/main/scala/akka/actor/CoordinatedShutdown.scala]: True

**Review by chbatey — APPROVED** [review]
LGTM

**Comment by chbatey** [comment]
needs a mima filter

**Comment by johanandren** [comment]
Fixes #24477

**Pull request merged**